### ISSUE TITLE
cucumber-rails: Rails 6 default for cache_classes generates a warning

### ISSUE THREAD

**Issue by fschwahn** [cucumber/cucumber-rails#458]
## Summary

In rails 6 application are now generated by default with `config.cache_classes = false` (see https://github.com/rails/rails/commit/65344f254cde87950c7f176cb7aa09c002a6f882). That generates a warning from cucumber-rails:

https://github.com/cucumber/cucumber-rails/blob/86a7f1f3f4943aae91239990b446f17699a769eb/lib/cucumber/rails.rb#L16-L20

## Expected Behavior

Adding cucumber-rails to a default rails app should ideally not create a warning.

## Current Behavior

A warning is printed when loading cucumber-rails. I think the real question is if that warning is still relevant? It's several years old by now.

**Comment by luke-hill** [comment]
Valid question. If you can reference some old SHA's / rails guides and let us know the best practices we can update this field.

**Comment by fschwahn** [comment]
I was able to track this back to this (prehistoric) commit, back when rails support was still in cucumber-core.
https://github.com/cucumber/cucumber-ruby/commit/e9ba91b94f91956d8c010deb45e334659d47e097

The `use_transactional_fixtures`-feature has been removed from cucumber-rails since 2011. Since then `database_cleaner` has been used for this (I think). I searched the issue tracker of database_cleaner, but could only find one issue which mentions `cache_classes` - so I guess it's not really a problem with `database_cleaner` anymore?

Another interesting point: when run through spring, cache_classes is false anyway (see https://github.com/rails/spring/blob/ca86653ba6d784f0348d5dd0280e030dd690af19/lib/spring/application.rb#L98-L109) - spring has been the default for new apps for a long time now, so it _seems_ that is not causing issues anymore.

TLDR: I _think_ this warning has become obsolete sometime in the past, and has only been carried forward. So it should be safe to remove.

**Comment by fschwahn** [comment]
@luke-hill what do you think about not emitting this warning if rails >= 6.0.0 is used? I could put together a PR for this.

**Comment by luke-hill** [comment]
@fschwahn Sorry I missed this. That seems perfectly valid. Can you just link me (Or better yet, reference in the changelog mention), to where this value is defaulted in rails6, then all is well

**Comment by fschwahn** [comment]
@luke-hill https://github.com/rails/rails/commit/65344f254cde87950c7f176cb7aa09c002a6f882 - is that enough? The changelog entry is also in there.

**Comment by luke-hill** [comment]
Perfect link. Can you add that when you write your changelog.

**Comment by luke-hill** [comment]
Is this something you're currently working on in a branch @fschwahn - If you can get it pushed up in the next couple of days I can review and turn it around quickly.

**Comment by fschwahn** [comment]
I just saw https://github.com/cucumber/cucumber-rails/pull/462 - that also fixes the issue by checking if spring is defined. `config.cache_classes` is only set to `false` if spring is enabled (as seen in the commit I referenced above).

So it might still be worth it to investigate if this warning has any merit at all, but the immediate issue with regards to a fresh default rails6 installations should be fixed.
